inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_zyx_fsv16_imad.cpp

```diff
@@ -194,44 +197,60 @@ Convolution_kernel_b_fs_yx_fsv16_imad::GetBlockParams(const convolution_params&
         }
     }
 
-    // Select biggest block height that fits into registers
+    // Select biggest block height and depth that fits into registers
     size_t block_height = 1;
+    size_t block_depth = 1;
     size_t in_block_height = 1;
-    for (size_t h = 2; h < 16; ++h) {
-        if (params.output.Y().v % h != 0)
-            continue;
+    size_t in_block_depth = 1;
 
-        size_t tmp_in_block_height = (h - 1) * params.stride.y + (params.filterSize.y - 1) * params.dilation.y + 1;
-        auto tmp_params = BlockParams{ block_width, h, block_features, in_block_width, tmp_in_block_height, feature_slm_split };
-
-        bool c_reg_pressure = EstimateRegPressure(params, tmp_params) <= max_reg_pressure;
-        bool c_occupancy = EstimateOccupancy(params, tmp_params) >= 1.f;
-        bool c_slm = EstimateSLMUsage(params, tmp_params) <= 1.f;
+    bool break_external_loop = false;
+    
+    for (size_t d = 2; d < 16; ++d) {
```

Comment:
Maybe this loop should start from 1? If d is a prime number and d > 16 (or any other case that prevents blocking by depth), then block_height will also remain equal to 1 that might be not optimal

Reply:
Fixed